### PR TITLE
Updated help descriptions of the commands

### LINKED ISSUE #48: Add help descriptions
For all commands add a description (from spec ;) )

### PR DESCRIPTION
**Pull request by ilayda21**
# Done
- Commands, arguments and options descriptions are updated.
- Added "Summarised Description" to every command and options to the [spec](https://docs.google.com/document/d/1zXq6ZtXi6UCUVOyF4_Fux13nDjePZWv-_JeEo0vytT0/edit#).

# QA
- Checkout the feature branch
- Install dependencies with `yarn`
- Run `ts-node ./src/index.ts --help` 
- Verify the program description is correct.
- Verify the usage is correct.
- Verify the "add-posts" command's description is correct.
- Verify the "delete-posts" command's description is correct.
- Run `ts-node ./src/index.ts add-posts --help`
- Verify the command's description is correct
- Verify the argument and options descriptions are correct.
- Verify the usage is correct. 
- Run `ts-node ./src/index.ts delete-posts --help`
- Verify the command's description is correct
- Verify the argument and options descriptions are correct.
- Verify the usage is correct.

Fixes #48

### CODE REVIEW

File: src/common/commandDescriptios.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+export const GENERAL_DESCRIPTION = "Greenhouse is a command-line tool that provides helpers to automate interactions with the Canonical Greenhouse website.";
+
+export const ADD_POSTS_USAGE = "([-i | --interactive] | <job-id> --regions=<region-name>[, <region-name-2>...] [--clone-from=<job-post-id>]) \n\n Examples: \n\t greenhouse add-posts --interactive \n \t greenhouse add-posts 1234 --regions=emea,americas \n \t greenhouse add-posts 1234 --regions=emea --clone-from=1123";
+
+export const ADD_POSTS_DESCRIPTION = "This command creates job posts for a specific job <job-id> in the specified regions <region-name> from all existing job posts in the Canonical Board. If the clone-from is specified, the command will create job posts in the specified regions from that job post. The command can be run interactively with the –interactive flag. This will prompt a set of questions for the user to answer.";
```
Comment (andesol) [inline src/common/commandDescriptios.ts:5]: This sentence sounds a bit repetitive to me:
"If the clone-from is specified, the command will create job posts in the specified regions from that job post"

Maybe something along the lines:
"If clone-from is specified, the new job post will be created based on the existing one"

(Can't come up with anything better right now)

File: src/common/commandDescriptios.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+export const ADD_POSTS_REGIONS_DESCRIPTION = "Mandatory argument that is the list of regions where the job posts will be created. The list of regions are: americas, emea, apac, us-boston";
+
+export const JOB_ID_DESCRIPTION = "The job-id is the ID of a job that the user is a Hiring lead of. This ID can be found in the URL of the job page. If the user is not a hiring lead on the specified job, an error message will be thrown. Example: https://canonical.greenhouse.io/sdash/12345 -> The job ID is 12345";
```
Comment (andesol) [inline src/common/commandDescriptios.ts:9]: It makes sense to lower case "hiring lead", IMHO

File: src/index.ts [hunk 3]
```diff
@@ -4,8 +4,9 @@ import Job from "./automations/Job";
 import SSO from "./automations/SSO";
 import { MAIN_URL, PROTECTED_JOB_BOARDS } from "./common/constants";
 import { joinURL } from "./common/pageUtils";
+import { ADD_POSTS_DESCRIPTION, ADD_POSTS_REGIONS_DESCRIPTION, ADD_POSTS_USAGE, CLONE_FROM_DESCRIPTION, DELETE_POSTS_DESCRIPTION, DELETE_POSTS_REGIONS_DESCRIPTION, DELETE_POSTS_USAGE, GENERAL_DESCRIPTION, INTERACTIVE_DESCRIPTION, JOB_ID_DESCRIPTION, SIMILAR_DESCRIPTION } from "./common/commandDescriptios";
```
Comment (andesol) [inline src/index.ts:7]: One little typo:

```suggestion
import { ADD_POSTS_DESCRIPTION, ADD_POSTS_REGIONS_DESCRIPTION, ADD_POSTS_USAGE, CLONE_FROM_DESCRIPTION, DELETE_POSTS_DESCRIPTION, DELETE_POSTS_REGIONS_DESCRIPTION, DELETE_POSTS_USAGE, GENERAL_DESCRIPTION, INTERACTIVE_DESCRIPTION, JOB_ID_DESCRIPTION, SIMILAR_DESCRIPTION } from "./common/commandDescriptions";
```

Alternatively you can do:
```
import * as commands from ...
```
As you prefer

**Review by andesol — APPROVED** [review]
LGTM!

Just some minor comments

**Review by tbille — CHANGES_REQUESTED** [review]
I just qa on the terminal, the text is super heavy :scream: 
I'll block the PR to have a chat with you if we can write down shorter summaries for the arguments.

**Comment by tbille** [comment]
@ilayda21 why is the usage always showing "index", I am guessing it's the name of the file. When we build the snap will it show index too or will it show greenhouse?

**Comment by ilayda21** [comment]
@tbille no it will display "greenhouse" instead of "index"

**Review by tbille — APPROVED** [review]
Looks good!

File: src/index.ts [hunk 4]
```diff
@@ -37,14 +37,12 @@ async function getJobPostInteractive(posts: PostInfo[], message: string) {
...
-                (post) =>
-                    post.boardInfo.name.toUpperCase() === board 
-            )
+            .filter((post) => post.boardInfo.name.toUpperCase() === board)
             .map((post) => post.name)
     );
 
-    if (!uniqueNames.size) throw Error(`No job post found in the Canonical board.`);
+    if (!uniqueNames.size)
+        throw Error(`No job post found in the Canonical board.`);
```
[the remark this answers is not in the document]
Comment (ilayda21) [reply src/index.ts:45]: yes if its single line you dont have to put them

**Pull request merged**